### ISSUE TITLE
Request to add Conan.io (Bintray) as a publish destination in CI systems

### ISSUE THREAD

**Issue by solvingJ** [Microsoft/cpprestsdk#479]
For those unfamiliar, Conan.io is a cross-platform package manager for C++ (similar to VCPKG) that has recently joined with Bintray.  Several people have already built CPPRest and published it to Conan, you can see the list by searching at this page: 
https://bintray.com/conan/conan-transit

These are unverified versions of the package.   It would be really great if the maintainers of this project setup Conan.io as a publish destination in the CI systems.  This way the community could have a verified official build of CPPRest from Microsoft in Conan.io,  just like we have for Java (in Maven Central) and .NET (in Nuget).  

There's an additional factor as well.  Swagger-codegen currently has a generator based on CPPRest.  It is currently generating the following unfortunate code inside the CMakeLists.txt to help users manually resolve the dependencies:
```
# Update require components as necessary
#find_package(Boost 1.45.0 REQUIRED COMPONENTS ${Boost_THREAD_LIBRARY} ${Boost_SYSTEM_LIBRARY} ${Boost_REGEX_LIBRARY} ${Boost_DATE_TIME_LIBRARY} ${Boost_PROGRAM_OPTIONS_LIBRARY} ${Boost_FILESYSTEM_LIBRARY})

# build and set path to cpp rest sdk
#set(CPPREST_ROOT ${PROJECT_SOURCE_DIR}/../../../developmentTools/3rdParty/cpprest)
set(CPPREST_INCLUDE_DIR ${CPPREST_ROOT}/include)
set(CPPREST_LIBRARY_DIR ${CPPREST_ROOT}/lib)

if( NOT DEFINED CPPREST_ROOT )
    message( FATAL_ERROR "Failed to find cpprest SDK (or missing components). Double check that \"CPPREST_ROOT\" is properly set")
endif( NOT DEFINED CPPREST_ROOT )
```

All the other major generators for Swagger-codegen have REST library dependency chains of their own, so they generate build system files with references to the common package managers for those dependencies (Maven/Nuget/NPM/Etc).  The generated libraries work as-generated without needing manual modifications to get them to work. 

An official and verified version of CPPRest on Conan would allow Swagger-codegen to leverage Conan for automatic dependency download and generate fully-functional SDK's based on CPPRest **with confidence**.  While this is possible today with the unverified builds of CPPRest, it's very undesirable and does not come with the same confidence that an official build would.

**Comment by ras0219-msft** [comment]
Hi @solvingJ,

Thanks for your suggestion.

In the current master branch, we've added `find_package()` support for cpprestsdk, which will make that unfortunate code into a simple:
```cmake
find_package(cpprestsdk REQUIRED)
```
In the meantime, they should instead use the standard cmake `find_XXX` commands
```cmake
find_path(CPPREST_INCLUDE_DIR include/cpprest/http_client.h)
find_library(CPPREST_LIBRARY NAMES cpprest_2_9d cpprest_2_9)
```
We believe this is the right way for projects like Swagger to consume cpprestsdk because it leaves choice in the hands of the user about how they want to acquire the library (apt/brew/vcpkg/conan/hunter/cget/etc).

It sounds like Conan is adding a curated repository that individuals can step up and maintain [1]. While we do not currently plan to maintain a C++REST SDK package there, I believe they will be accepting contributions from interested individuals who are willing to sign up as maintainers. A properly constructed package should be compatible with the above `find_package()` command.

[1] http://blog.conan.io/2017/05/25/bintray-support-conan-repositories.html

**Comment by solvingJ** [comment]
@ras0219-msft 
I just discovered that you are a core maintainer of both this CPPRest and the VCPKG projects.  Despite VCPKG and Conan.io not quite being exactly the same in their goals, they're close enough that they're in some level of competition with each other.  Thus, in retrospect it felt pretty silly to have asked you to build and package your library for your competitor.  It felt even more silly when I saw that you personally took the time to build the packaging for [CPPRest for VCPKG](https://github.com/Microsoft/vcpkg/tree/master/ports/cpprestsdk).  I apologize for my ignorance there, and thanks for all the work you've done. 

With that said, I'm glad to see C++ getting a lot of attention around package management, even if it's a little fragmented at the moment.  In any case, an effective, widely adopted, cross-platform package management system will be a huge step forward in reducing barriers for new C++ developers.  

I still maintain that it would be great if CPPRest followed the lead of the Microsoft Autorest team, making ease-of-consumption by the C++ community a top priority, and deploying to the predominant package manager/repository for the language.  For Autorest, their languages had more established package managers than Conan.io, and also did not have the same unfortunate conflict of interest that exists here, so their decisions were quite a bit simpler.  Nonetheless, I will continue to hold out hope that the CPPRest project can one day find the objectivity to see Conan.io not as a competitor to be shunned, but rather as one of the best channels available for empowering developers to include the library in modern C++ projects of the future.  Thanks again for the contributions.
